sdks/python/apache_beam/ml/transforms/handlers.py

```diff
@@ -134,7 +135,9 @@ def process(self, element):
         hash_object.update(str(list(value)).encode())
       else:  # assume value is a primitive that can be turned into str
         hash_object.update(str(value).encode())
-    yield (hash_object.hexdigest(), element)
+    # add a unique suffix to the hash key to avoid collisions.
+    unique_suffix = uuid.uuid4().hex
```

Comment:
Thinking about this, we can just attach a unique id to each element. We just need unique id so that we can group transformed pcoll with untransformed pcoll with the downstream CoGroupByKey Pcoll

Reply:
> Thinking about this, we can just attach a unique id to each element.

Not sure how accurate this is:
https://stackoverflow.com/questions/72989272/python-generates-same-uuid-over-multiple-docker-containers
  but I was also thinking about how the seed will be initialized in docker context.

and UUID1 might be not thread-safe:
https://github.com/ClickHouse/clickhouse-connect/issues/194

I'd expect the combination of   uuid1 + os PID + uuid4  would be exceedingly unlikely to collide in beam/dataflow context even with the above consideration. we can and should detect collisions though, we can fail the pipeline if that happens.

A few other notes:

1. Please add a known issue to Changes.MD and file a GH issue that will be fixed by this PR.
2. Let's rename the ComputeAndAttachHashKey to ComputeAndAttachUniqueID
3. I wonder if performance and pipeline cost would improve if we  can find a way to pass-through columns that do not need to be processed to tft, converting them to bytes if necessary.

Reply:
also i wonder what is the overhead of TFTransform compared to plain usage of TFT, although that's a separate question

Reply:
also if keys can be `bytes` instead of `str`, using raw bytes instead of hash would cut key size in ~half

Reply:
I think we shouldn't raise an error as of now since the chance of collision is very low.
 
>> I wonder if performance and pipeline cost would improve if we can find a way to pass-through columns that do not need to be processed to tft, converting them to bytes if necessary, avoiding the shuffle step.

I can try this and test the performance. If this works, we can remove the CoGroupByKey altogether.